### ISSUE TITLE
Ecto cannot cast DateTime in query

### ISSUE THREAD

**Issue by kurayama** [lau/calecto#12]
Getting this error when using a Calendar.DateTime in a where:

Repo.all(from t in Tactic, where: t.account_id == ^account_id and t.updated_at >= ^sinceb, select: t)
*\* (UndefinedFunctionError) undefined function: Calendar.DateTime.type/0
    (ecto) lib/ecto/query/planner.ex:124: Ecto.Query.Planner.prepare/3
    (ecto) lib/ecto/query/planner.ex:66: Ecto.Query.Planner.query/4
    (ecto) lib/ecto/repo/queryable.ex:91: Ecto.Repo.Queryable.execute/5
    (ecto) lib/ecto/repo/queryable.ex:15: Ecto.Repo.Queryable.all/4

**Comment by rockwood** [comment]
@kurayama What happens if you explicitly corse the value to with the `type` function?

```
Repo.all(from t in Tactic, where: t.account_id == ^account_id and t.updated_at >= type(^sinceb,  Calecto.DateTimeUTC), select: t)
```

**Comment by kurayama** [comment]
Same error, it only works if sinceb is a map instead of a Calendar.DateTime
Like {"day" => 1, "hour" => 1, "min" => 12, "month" => 12, "sec" => 0, "timezone" => "Etc/UTC", "usec" => 0, "year" => 2015}

**Comment by lau** [comment]
This needs to be easier. I have an idea for a better solution, but it won't be ready before tomorrow. I'll let you know.

**Comment by lau** [comment]
@kurayama I have released Calecto 0.4.0. With that you should be able to cast a Calendar.DateTime to Calecto.DateTimeUTC and use that to compare in an Ecto query.

Example: `{:ok, sinceb} = Calendar.DateTime.now_utc |> Calecto.DateTimeUTC.cast`

Another example with parsing a unix time and then shifting it using calendar: `{:ok, sinceb} = Calendar.DateTime.Parse.unix!(1440000123) |> Calendar.DateTime.advance!(-86400) |> Calecto.DateTimeUTC.cast`

Let me know if it solves your problem.
